bddtests/common_steps.go

```diff
@@ -1168,31 +1211,107 @@ func (d *CommonSteps) httpGet(url string) error {
 
 	resp, err := client.Get(url)
 	if err != nil {
-		return err
+		return nil, 0, err
 	}
 
-	if resp.StatusCode != http.StatusOK {
-		return errors.Errorf("received status code %d", resp.StatusCode)
+	payload, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		return nil, 0, fmt.Errorf("reading response body failed: %s", err)
 	}
 
 	contentType, ok := resp.Header["Content-Type"]
 	if ok && strings.HasPrefix(contentType[0], "image") {
 		logger.Infof("... got HTTP image of type [%s]", contentType)
-		return nil
+	} else if ok {
+		logger.Infof("... got HTTP response of type [%s]:\n%s", contentType[0], payload)
+	} else {
+		logger.Infof("... got HTTP response of unknown type:\n%s", payload)
 	}
 
-	payload, err := ioutil.ReadAll(resp.Body)
+	return payload, resp.StatusCode, nil
+}
+
+func (d *CommonSteps) httpPost(url, path string) error {
+	resp, code, err := d.doHTTPPost(url, path)
 	if err != nil {
-		return fmt.Errorf("reading response body failed: %s", err)
+		return err
 	}
 
-	logger.Infof("... got HTTP response of type [%s]:\n%s", contentType[0], payload)
+	SetResponse(string(resp))
 
-	SetResponse(string(payload))
+	if code != http.StatusOK {
+		return errors.Errorf("received status code %d", code)
+	}
 
 	return nil
 }
 
+func (d *CommonSteps) httpPostWithExpectedCode(url, path string, expectingCode int) error {
+	resp, code, err := d.doHTTPPost(url, path)
+	if err != nil {
+		return err
+	}
+
+	SetResponse(string(resp))
+
+	if code != expectingCode {
+		return errors.Errorf("expecting status code %d but got %d", expectingCode, code)
+	}
+
+	logger.Infof("Returned status code is %d which is the expected status code", code)
+
+	return nil
+}
+
+func (d *CommonSteps) doHTTPPost(url, path string) ([]byte, int, error) {
+	logger.Infof("Uploading file [%s] to [%s]", path, url)
+
+	fileBytes := getFile(path)
+
+	client := &http.Client{}
+
+	if strings.HasPrefix(url, "https") {
+		// TODO add tls config https://github.com/trustbloc/fabric-peer-test-common/issues/51
+		// TODO !!!!!!!remove InsecureSkipVerify after configure tls for http client
+		client.Transport = &http.Transport{
+			TLSClientConfig: &tls.Config{InsecureSkipVerify: true}, //nolint: gosec
```

Comment:
We need to fix this.

Reply:
@fqutishat @bstasyszyn

Reply:
Should be a separate PR. We have these TODOs in a number of spots.